File: 23단계/2606.py
import sys

def buildGraph():
    global adj
    E = int(sys.stdin.readline().strip())
    for i in range(E):
        s,d = map(int,sys.stdin.readline().strip().split())
        adj[s-1].append(d-1)
        adj[d-1].append(s-1)

def dfs(idx):
    global adj, visited, cnt
    
    visited[idx] = True
    cnt +=1
    for i in adj[idx]:
        if visited[i] == False:
            dfs(i)
            
N = int(sys.stdin.readline().strip())
global adj, visited, cnt
cnt = 0
adj = list([] for i in range(N))
visited = list(False for i in range(N))
buildGraph()
dfs(0)
print(cnt-1)

File: Archived/23단계/2606.java
import java.util.*;
import java.io.*;

public class Main{
    static BufferedReader br;
    static int V,E,cnt;
    static boolean[] visited;
    static ArrayList<ArrayList<Integer>> adj;
    public static void main(String[] args) throws IOException{
        br = new BufferedReader(new InputStreamReader(System.in));
        String[] tmp; int s,d;
        adj = new ArrayList<>();
        V = Integer.parseInt(br.readLine());
        E = Integer.parseInt(br.readLine());
        visited = new boolean[V];
        for(int i=0;i<V;i++)
            adj.add(new ArrayList<>());
        for(int i=0;i<E;i++){
            tmp = br.readLine().split(" ");
            s = Integer.parseInt(tmp[0])-1;
            d = Integer.parseInt(tmp[1])-1;
            adj.get(s).add(d);
            adj.get(d).add(s);
        }
        dfs(0);
        System.out.println(cnt-1);
    }
    public static void dfs(int idx){
        visited[idx] = true;
        cnt++;
        for(int i:adj.get(idx)){
            if(!visited[i]){
                dfs(i);
            }
        }
    }
}